### PR TITLE
Add an example of how to create a memfile and write data to it and read it again

### PR DESCRIPTION
**Pull request by mrhdias**
There is nowhere in the documentation how to create a file and add data and read it again.

### CODE REVIEW

File: lib/pure/memfiles.nim [hunk 1]
```diff
@@ -15,6 +15,27 @@
 ## It also provides some fast iterators over lines in text files (or
 ## other "line-like", variable length, delimited records).
 
+runnableExamples("-r:off"):
+  # Example of how to create a memfile and
+  # write data to it and read it again.
```
Comment (ghost) [inline lib/pure/memfiles.nim]:
```suggestion
  # An example of how to create a memfile,
  # write data to it and read it again. 
```

File: lib/pure/memfiles.nim [hunk 2]
```diff
@@ -15,6 +15,27 @@
 ## It also provides some fast iterators over lines in text files (or
 ## other "line-like", variable length, delimited records).
 
+runnableExamples("-r:off"):
+  # Example of how to create a memfile and
+  # write data to it and read it again. 
+  import memfiles, os
```
Comment (ghost) [inline lib/pure/memfiles.nim]:
```suggestion
  import std/[memfiles, os]
```

but actually see 
https://github.com/nim-lang/Nim/pull/17203
https://github.com/nim-lang/RFCs/issues/352

about including the module's own import in the top-level example

File: lib/pure/memfiles.nim [hunk 3]
```diff
@@ -15,6 +15,27 @@
...
+
+  let fn = "/tmp/test.mmap"
```
Comment (ghost) [inline lib/pure/memfiles.nim]:
```suggestion
  let fn = "test.mmap"
```

That way it's more portable

Comment (mrhdias) [reply lib/pure/memfiles.nim]: @Yardanico, your changes are fine for me, thanks a lot.

Comment (timotheecour) [reply lib/pure/memfiles.nim]: use this instead:
```nim
let fn = getTempDir() / "test.mmap"
```

File: lib/pure/memfiles.nim [hunk 4]
```diff
@@ -15,6 +15,27 @@
...
+  if fileExists(fn): removeFile(fn)
+
+  var mm = open(fn, mode = fmWrite, newFileSize = 20)
+
+  let data = "Hello".cstring
+  copyMem(mm.mem, cast[pointer](data), data.len+1)
+  mm.close()
+
+  var mm_full = memfiles.open(fn, mode = fmRead)
+  echo "size:", mm_full.size
```
Comment (timotheecour) [inline lib/pure/memfiles.nim]: assert is more informative:
```suggestion
  assert mm_full.size == data.len + 1 # (or fix as needed)
```

File: lib/pure/memfiles.nim [hunk 5]
```diff
@@ -15,6 +15,27 @@
...
+  echo cast[cstring](mm_full.mem)
```
Comment (timotheecour) [inline lib/pure/memfiles.nim]: ditto, use assert, not echo:
```nim
assert cast[cstring](mm_full.mem) == data
```

File: lib/pure/memfiles.nim [hunk 6]
```diff
@@ -15,6 +15,27 @@
...
+
+  if fileExists(fn): removeFile(fn)
+
+  var mm = open(fn, mode = fmWrite, newFileSize = 20)
+
+  let data = "Hello".cstring
+  copyMem(mm.mem, cast[pointer](data), data.len+1)
+  mm.close()
+
+  var mm_full = memfiles.open(fn, mode = fmRead)
```
Comment (timotheecour) [inline lib/pure/memfiles.nim:33]: can this be let?

File: lib/pure/memfiles.nim [hunk 7]
```diff
@@ -15,6 +15,27 @@
...
 ## other "line-like", variable length, delimited records).
 
+runnableExamples("-r:off"):
+  # Example of how to create a memfile and
+  # write data to it and read it again. 
+  import memfiles, os
+
+  let fn = "/tmp/test.mmap"
+
+  if fileExists(fn): removeFile(fn)
```
Comment (timotheecour) [inline lib/pure/memfiles.nim]:
```suggestion
  removeFile(fn)
```
(see docs: `This does not fail if the file never existed in the first place.`)

File: lib/pure/memfiles.nim [hunk 8]
```diff
@@ -15,6 +15,27 @@
...
+
+  var mm = open(fn, mode = fmWrite, newFileSize = 20)
```
Comment (timotheecour) [inline lib/pure/memfiles.nim:27]: can this be let?

File: lib/pure/memfiles.nim [hunk 9]
```diff
@@ -15,6 +15,27 @@
 ## It also provides some fast iterators over lines in text files (or
 ## other "line-like", variable length, delimited records).
 
+runnableExamples("-r:off"):
```
Comment (timotheecour) [inline lib/pure/memfiles.nim]:
```suggestion
runnableExamples:
```

better to actually run the test if possible

File: lib/pure/memfiles.nim [hunk 10]
```diff
@@ -15,6 +15,27 @@
 ## It also provides some fast iterators over lines in text files (or
 ## other "line-like", variable length, delimited records).
 
+runnableExamples:
+  # An example of how to create a memfile,
+  # write data to it and read it again. 
+  import memfiles, os
+
+  let fn = "/tmp/test.mmap"
```
Comment (Araq) [inline lib/pure/memfiles.nim]:
```suggestion
  let fn = getTempDir() / "test.mmap"
```

File: lib/pure/memfiles.nim [hunk 11]
```diff
@@ -15,6 +15,27 @@
...
+
+  var mm = open(fn, mode = fmWrite, newFileSize = 20)
+
+  let data = "Hello".cstring
+  copyMem(mm.mem, cast[pointer](data), data.len+1)
+  mm.close()
+
+  var mm_full = memfiles.open(fn, mode = fmRead)
+  echo "size:", mm_full.size
+  echo cast[cstring](mm_full.mem)
```
Comment (Araq) [inline lib/pure/memfiles.nim]:
```suggestion
  assert cast[cstring](mm_full.mem) == data
```

File: lib/pure/memfiles.nim [hunk 12]
```diff
@@ -15,6 +15,27 @@
 ## It also provides some fast iterators over lines in text files (or
 ## other "line-like", variable length, delimited records).
 
+runnableExamples:
+  # An example of how to create a memfile,
+  # write data to it and read it again. 
+  import std/[memfiles, os]
```
Comment (konsumlamm) [inline lib/pure/memfiles.nim:21]:
```suggestion
  import std/os
```
`import std/memfiles` is automatically added at the top.

**Pull request closed without merging**